### PR TITLE
add course exclusion on curriculum course membership

### PR DESCRIPTION
**Pull request by jansenk**
screen shot 2019-02-26 at 10 09 42 am

### CODE REVIEW

File: course_discovery/apps/course_metadata/models.py [hunk 1]
```diff
@@ -1768,11 +1768,24 @@ class CurriculumCourseMembership(TimeStampedModel):
     """
     curriculum = models.ForeignKey(Curriculum, on_delete=models.CASCADE)
     course = models.ForeignKey(Course, on_delete=models.CASCADE)
+    course_run_exclusions = models.ManyToManyField(
+        CourseRun, through='course_metadata.CurriculumCourseRunExclusion', related_name='excluded_curricula'
```
Comment (iloveagent57) [inline course_discovery/apps/course_metadata/models.py]: The `related_name` should be something that refers back to this model from the related model, e.g. `curriculum_course_membership`.

https://docs.djangoproject.com/en/1.11/ref/models/fields/#django.db.models.ForeignKey.related_name

**Review by iloveagent57 — CHANGES_REQUESTED** [review]
Just gotta change that `related_name` value, the rest looks good!

**Review by schenedx — APPROVED** [review]
👍

File: course_discovery/apps/course_metadata/models.py (hunk 1 above)
Comment (iloveagent57) [reply course_discovery/apps/course_metadata/models.py]: Oh, man, I'm an idiot.  You wanted to indicate from the CourseRun object that the set of things you're getting back are curricula that the run is _excluded_ from, huh?  Sorry about that.  I'm gonna check out your branch on my machine to see how this looks in the shell, but I think your first attempt was correct: you want to mention somewhere in the related_name the idea of exclusion.

Comment (iloveagent57) [reply course_discovery/apps/course_metadata/models.py]: screen shot 2019-02-26 at 3 13 02 pm
OK, I checked out your branch and created an exclusion (see screenshot above).
In the Django shell, I can now refer to it's related objects like this:

```
>>> from course_discovery.apps.course_metadata.models import *
>>> CourseRun.objects.last()
<CourseRun: course-v1:edX+DemoX+Demo_Course: edX Demonstration Course>
>>> cr.curriculum_course_membership.all()
<QuerySet [<CurriculumCourseMembership: CurriculumCourseMembership object>]>
>>> cr.curriculum_course_membership.first()
<CurriculumCourseMembership: CurriculumCourseMembership object>
>>> cr.curriculum_course_membership.first().curriculum
<Curriculum: e629c095-3805-4c72-95df-5fbe5ed1e982>
>>> cr.curriculum_course_membership.first().course
<Course: edX+DemoX: edX Demonstration Course>
# That all actually looks fine...
# this related object manager is automagically named by Django
>>> cr.curriculumcourserunexclusion_set.all()
<QuerySet [<CurriculumCourseRunExclusion: CurriculumCourseRunExclusion object>]>
>>> cr.curriculumcourserunexclusion_set.first()
<CurriculumCourseRunExclusion: CurriculumCourseRunExclusion object>
>>> cr.curriculumcourserunexclusion_set.first().course_membership
<CurriculumCourseMembership: CurriculumCourseMembership object>
>>> cr.curriculumcourserunexclusion_set.first().course_run  # this should just be cr again
<CourseRun: course-v1:edX+DemoX+Demo_Course: edX Demonstration Course>
>>> cr == cr.curriculumcourserunexclusion_set.first().course_run
True
>>> # yup :)
```
So actually, I think how the related object naming worked out is just fine.  We can change it later with a migration if it's problematic.

**Review by iloveagent57 — APPROVED** [review]

**Pull request merged**